### PR TITLE
Don't apply `CountOnNullRector` to variable that are `array` for sure.

### PR DESCRIPTION
**Pull request by Aerendir**
Issue #1551.

### CODE REVIEW

**Comment by Aerendir** [comment]
The tests here passes, as if it is not problem.

But as per Issue, problems actually exist.

@TomasVotruba , please, can you check?

File: packages/Php/tests/Rector/FuncCall/CountOnNullRector/Fixture/local_variable.php.inc [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        // appear very similar to this.
+        foreach ($anArray as $value) {
+            if ('condition' === $value) {
+                $results = 'transform into a string';
+            } elseif (is_array($results)) {
+                $results[] = $value;
+            }
+        }
+
+        return (is_array($results) || $results instanceof \Countable ? count($results) : 0) > 0 ? $results : null;
```
Comment (stloyd) [inline packages/Php/tests/Rector/FuncCall/CountOnNullRector/Fixture/local_variable.php.inc]: What about using: `is_iterable($results)`?

File: packages/Php/tests/Rector/FuncCall/CountOnNullRector/Fixture/local_variable.php.inc [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    public function methodWithLocalVariableInitializedAsArrayAndTransformedIntoString(array $anArray): ?array
+    {
+        $results = [];
+
+        // This code may not have much sense,
+        // but in real world scenario it may
+        // appear very similar to this.
+        foreach ($anArray as $value) {
+            if ('condition' === $value) {
+                $results = 'transform into a string';
```
Comment (stloyd) [inline packages/Php/tests/Rector/FuncCall/CountOnNullRector/Fixture/local_variable.php.inc]: TBH. I'm not sure this should be supported at all cause string is not countable in newer PHP versions, yet this probably will fail while checking code with phpstan.

**Comment by TomasVotruba** [comment]
@Aerendir Could you maybe add just the minimal example that broke your code? 4-5 lines shold do.

It's too long to review and with no added value.

File: packages/Php/tests/Rector/FuncCall/CountOnNullRector/Fixture/local_variable.php.inc (hunk 1 above)
Comment (Aerendir) [reply packages/Php/tests/Rector/FuncCall/CountOnNullRector/Fixture/local_variable.php.inc]: This is not something I can decide: I'm simply following the way the rector works.

To change it, a change in the rector, too, is required.

ping @TomasVotruba

File: packages/Php/tests/Rector/FuncCall/CountOnNullRector/Fixture/local_variable.php.inc (hunk 2 above)
Comment (Aerendir) [reply packages/Php/tests/Rector/FuncCall/CountOnNullRector/Fixture/local_variable.php.inc]: Yes, you are right. In my opinion, it is something that should be avoided at all.

I added it for completeness, but there's nothing against removing it.

ping @TomasVotruba

**Comment by Aerendir** [comment]
@TomasVotruba , I'll try, but the test I've submitted already contains the failing code: is the method `methodWithLocalVariableArrayAsResult()`: in my code it is changed while in this test it isn't...

File: packages/Php/tests/Rector/FuncCall/CountOnNullRector/Fixture/local_variable.php.inc (hunk 1 above)
Comment (TomasVotruba) [reply packages/Php/tests/Rector/FuncCall/CountOnNullRector/Fixture/local_variable.php.inc]: @stloyd Why not. Could you show me the compatibility with https://3v4l.org ?

**Comment by TomasVotruba** [comment]
The smaller the code the easier to find why

**Comment by TomasVotruba** [comment]
Closing in favor of #1563

**Pull request closed without merging**